rdflib/collection.py

```diff
@@ -243,6 +250,9 @@ def append(self, item: Node) -> Collection:
         return self
 
     def __iadd__(self, other: Iterable[Node]):
+        if self.uri == RDF.nil:
+            raise ValueError("Cannot append to empty list")
+
         end = self._end()
```

Comment:
I was thinking like this:
```suggestion
        end = self._end()
        if end == RDF.nil:
            raise ValueError("Cannot append to empty list")
```

I think `_end()` will still return `nil` when `self.uri` is `nil`, this is what we want.

Note, I think you can append `nil` to the `rdf:rest` on the end of a populated list, that will cause it to be a finalized list, that cannot have more items added.

So if `_end()` returns `rdf:nil` then the list is either empty or finalized, either way it cannot have another item appended.